guides/plugins/plugins/framework/rate-limiter/add-rate-limiter-to-api-route.md

```diff
@@ -0,0 +1,147 @@
+
+Each rate limit configuration needs the following keys:
+- `enabled`: Enables / Disables the rate limit for the specific route (default value: true).
+- `policy`: Possible policies are `fixed_window`, `sliding_window`, `token_bucket`, `time_backoff`. For more information check the [Symfony documentation](https://symfony.com/doc/current/rate_limiter.html#rate-limiting-policies).
+
+If you plan to configure the `time_backoff` policy, head over to [the next part](#configuring-time-backoff-policy).
+Otherwise, check the [Symfony documentation](https://symfony.com/doc/current/rate_limiter.html#configuration) for the other keys you need for each policy.
+
+{% code title="<platform root>/src/Core/Framework/Resources/config/packages/shopware.yaml" %}
+```yaml
+shopware:
+    api:
+        rate_limiter:
+            example_route:
+                enabled: true
+                policy: 'time_backoff'
+```
+{% endcode %}
+
+### Configuring time backoff policy
+
+The `time_backoff` policy is build by Shopware itself. It enables you to throttle the request in multiple steps with different waiting times.
+In this example we're going to configure a limit, which throttles the user for 10 seconds after 3 requests and starting from 5 requests it always
+throttles the request for 60 seconds.
+
+{% code title="<platform root>/src/Core/Framework/Resources/config/packages/shopware.yaml" %}
+```yaml
+shopware:
+    api:
+        rate_limiter:
+            example_route:
+                enabled: true
+                policy: 'time_backoff'
+                reset: '24 hours'
+                limits:
+                    - limit: 3
+                      interval: '10 seconds'
+                    - limit: 5
+                      interval: '60 seconds'
+```
+{% endcode %}
+
+## Implementing rate limit in API route
+
+### Inject service
+
+After we've configured our rate limit, we want to use it in our API route.
+For this we need to inject the `Shopware\Core\Framework\RateLimiter\RateLimiter` service.
+
+{% code title="<plugin root>/src/Core/Content/Example/SalesChannel/ExampleRoute.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Core\Content\Example\SalesChannel;
+
+use Shopware\Core\Framework\RateLimiter\RateLimiter;
+...
+
+/**
+ * @RouteScope(scopes={"store-api"})
+ */
+class ExampleRoute extends AbstractExampleRoute
+{
+    private RateLimiter $rateLimiter;
+
+    public function __construct(RateLimiter $rateLimiter)
+    {
+        $this->rateLimiter = $rateLimiter;
+    }
+
+    ...
+}
+```
+{% endcode %}
+
+### Call the rate limiter
+
+After we've injected the service into our API route, we cann call the limiter in our route method.
+
+To do this, we call the method `ensureAccepted` of the rate limiter which accepts the following arguments:
+- `route`: Unique name of the rate limit, we defined in the [configuration](#configuring-rate-limit).
+- `key`: Key with which we want to limit the request e.g., the client IP.
```

Comment:
Please speak that out loud quickly three times `with which we want` 😁 Maybe there is a better phrase for that